sqlmesh/core/test/definition.py

```diff
@@ -57,6 +58,7 @@ def __init__(
         path: Path | None = None,
         preserve_fixtures: bool = False,
         default_catalog: str | None = None,
+        lock: t.Optional[threading.Lock] = None,
```

Comment:
I don't like the idea of passing concurrency primitives from outside. This is asking for trouble.